clients/v1_auth/v1_auth.go

```diff
@@ -0,0 +1,441 @@
+	deleted bool
+	token   *v1Token
+	tokens  []v1Token
+}
+
+type v1Token struct {
+	ID          string   `json:"id"`
+	Description string   `json:"description"`
+	Token       string   `json:"token"`
+	Status      string   `json:"status"`
+	UserName    string   `json:"userName"`
+	UserID      string   `json:"userID"`
+	Permissions []string `json:"permissions"`
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	Desc        string
+	Username    string
+	Password    string
+	NoPassword  bool
+	ReadBucket  []string
+	WriteBucket []string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if params.Password != "" && params.NoPassword {
+		return fmt.Errorf("only one of --password and --no-password may be specified")
+	}
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	// verify an existing token with the same username doesn't already exist
+	auths, err := c.LegacyAuthorizationsApi.GetAuthorizations(ctx).Token(params.Username).Execute()
+	if apiError, ok := err.(api.ApiError); ok {
+		if apiError.ErrorCode() != api.ERRORCODE_NOT_FOUND {
+			return fmt.Errorf("failed to verify username %q has no auth: %w", params.Username, err)
+		} else if len(auths.GetAuthorizations()) != 0 {
+			return fmt.Errorf("authorization with username %q already exists", params.Username)
+		}
+	}
+
+	password := params.Password
+	if password == "" && !params.NoPassword {
+		pass, err := c.StdIO.GetPassword("Please type your password")
+		if err != nil {
+			return err
+		}
+		password = pass
+	}
+
+	bucketPerms := []struct {
+		action string
+		perms  []string
+	}{
+		{action: "read", perms: params.ReadBucket},
+		{action: "write", perms: params.WriteBucket},
+	}
+
+	var permissions []api.Permission
+	for _, bp := range bucketPerms {
+		for _, p := range bp.perms {
+			// verify the input ID
+			if _, err := influxid.IDFromString(p); err != nil {
+				return fmt.Errorf("invalid bucket ID '%s': %w (did you pass a bucket name instead of an ID?)", p, err)
+			}
+
+			newPerm := api.Permission{
+				Action: bp.action,
+				Resource: api.PermissionResource{
+					Type:  "buckets",
+					Id:    &p,
+					OrgID: &orgID,
+				},
+			}
+			permissions = append(permissions, newPerm)
+		}
+	}
+	authReq := api.LegacyAuthorizationPostRequest{
+		Description: &params.Desc,
+		OrgID:       orgID,
+		Permissions: permissions,
+		Token:       &params.Username,
+	}
+
+	newAuth, err := c.LegacyAuthorizationsApi.PostAuthorizations(ctx).LegacyAuthorizationPostRequest(authReq).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to create new authorization: %w", err)
+	}
+
+	if password != "" {
+		err := c.LegacyAuthorizationsApi.SetAuthorizationsIDPassword(ctx, newAuth.GetId()).
+			Password(password).
+			Execute()
+		if err != nil {
+			_ = c.LegacyAuthorizationsApi.DeleteAuthorizationsID(ctx, newAuth.GetId()).Execute()
+			return fmt.Errorf("failed to update password for user ID %q: %w", newAuth.GetUserID(), err)
```

Comment:
I think this error message would be more understandable for users:
```suggestion
			return fmt.Errorf("failed to set password for %q: %w", params.Username, err)
```